### PR TITLE
KPV-2320 :: Disabling buttons logic at vote and watching the contest application view.

### PR DESCRIPTION
**Pull request by pverdu-kuorum**
Now the proper buttons should be disabled when an user votes, considering if its known or anonymous.

Now the buttons should be disabled when a logged user visits the contest application page wich he have already voted.

### CODE REVIEW

File: web-app/js/campaigns/contestApplication.js [hunk 1]
```diff
@@ -62,14 +62,19 @@ var contestApplicationFunctions = {
                 if (contestApplicationVote.success) {
                     console.log("Vote Success")
                     console.log(contestApplicationVote)
```
Comment (iduetxe) [inline web-app/js/campaigns/contestApplication.js]: Remove this logs?

File: web-app/js/campaigns/contestApplication.js [hunk 2]
```diff
@@ -62,14 +62,19 @@ var contestApplicationFunctions = {
...
-                    contestApplicationFunctions._disableButton($buttonColumnC, contestApplicationVote.vote.votes);
-                    contestApplicationFunctions._disableButton($liStats, contestApplicationVote.vote.votes);
+                    if($button.attr("data-loggedUser") !== '') {
+                        contestApplicationFunctions._disableVoteButton($buttonMainPage, contestApplicationVote.vote.votes);
+                        contestApplicationFunctions._disableVoteButton($buttonColumnC, contestApplicationVote.vote.votes);
+                        contestApplicationFunctions._disableVoteButton($buttonContestApplicationList, contestApplicationVote.vote.votes);
+                    }
+                    contestApplicationFunctions._updateNumVotes($buttonMainPage, contestApplicationVote.vote.votes);
+                    contestApplicationFunctions._updateNumVotes($liStats, contestApplicationVote.vote.votes);
+                    contestApplicationFunctions._updateNumVotes($buttonContestApplicationList, contestApplicationVote.vote.votes);
```
Comment (iduetxe) [inline web-app/js/campaigns/contestApplication.js]: For improvung the readabilty, could be in a function called "updateButtonsAfterSuccess" or something like this

**Review by iduetxe — CHANGES_REQUESTED** [review]
It is a simple refactor for extracting a "private" function

File: web-app/js/campaigns/contestApplication.js (hunk 1 above)
Comment (pverdu-kuorum) [reply web-app/js/campaigns/contestApplication.js]: Removed

File: web-app/js/campaigns/contestApplication.js (hunk 2 above)
Comment (pverdu-kuorum) [reply web-app/js/campaigns/contestApplication.js]: Extracted to _handleButtonsAtSuccess function.

**Review by iduetxe — APPROVED** [review]

File: web-app/js/campaigns/contestApplication.js [hunk 3]
```diff
@@ -84,12 +74,35 @@ var contestApplicationFunctions = {
         });
     },
 
-    _disableButton: function ($button, numVotes) {
-        $button.attr("disabled", "disabled")
+    _handleButtonsAtSuccess: function (contestApplicationVote, $button) {
+        const $buttonMainPage = $("section#main .leader-post .contestApplication-vote")
+        const $buttonColumnC = $("#aside-ppal .call-to-action .actions .contestApplication-vote a")
+        const $buttonContestApplicationList = $("#contest-applications-list  .contestApplication-vote-" + contestApplicationVote.vote.contestApplicationId)
```
Comment (DavidSanFer555) [inline web-app/js/campaigns/contestApplication.js:80]: Better somthing like this to reduce copyPaste
```javascript
  _handleButtonsAtSuccess: function (contestApplicationVote, $button) {
        const $buttonMain = $buttonMainPage = $("section#main .leader-post .contestApplication-vote");
        const $buttonContestApplicationList = $("#contest-applications-list  .contestApplication-vote-" + contestApplicationVote.vote.contestApplicationId);
        const buttons = [$buttonMain, $buttonContestApplicationList,
            $("#aside-ppal .call-to-action .actions .contestApplication-vote a")]
        const stats = [$buttonMain,$buttonContestApplicationList,
            $(".leader-post-stats .fa-scroll").parents("li")]

        const numVotes = contestApplicationVote.vote.votes;
        if ($button.attr("data-loggedUser") !== '') {
            buttons.forEach(function (item){contestApplicationFunctions._disableVoteButton(item,numVotes)});
        }
        stats.forEach(function (item){contestApplicationFunctions._updateNumVotes(item,numVotes)});

    }
```

**Review by DavidSanFer555 — CHANGES_REQUESTED** [review]
Suggested javascript refactor

**Pull request merged**